src/plugins/intel_gpu/src/kernel_selector/kernels/mvn/mvn_kernel_bfyx_opt.cpp

```diff
@@ -37,52 +38,88 @@ MVNKernelBfyxOpt::Parent::DispatchData MVNKernelBfyxOpt::SetDefault(const mvn_pa
+    dispatchData.maxSlmSize = max_lws;
+    if (!params.has_dynamic_tensors()) {
+        if (params.mvnMode == MVNMode::WITHIN_CHANNELS) {
+            dispatchData.dataSetSize = input.X().v * input.Y().v * input.Z().v;
+            dispatchData.dataSetsCount = input.Batch().v * input.Feature().v;
+        } else {
+            dispatchData.dataSetSize = input.X().v * input.Y().v * input.Z().v * input.Feature().v;
+            dispatchData.dataSetsCount = input.Batch().v;
+        }
 
-    dispatchData.lws[0] = 1;
-    dispatchData.lws[1] = 1;
-    dispatchData.lws[2] = 1;
-    // Compute maximum possible LWS that does not exceed device capabilities and optimizes number of global memory
-    // reads.
-    while ((dispatchData.itemsNum > 32 || dispatchData.lws[0] < dispatchData.itemsNum) && (2 * dispatchData.lws[0] <= max_lws)) {
-        dispatchData.lws[0] *= 2;
-        dispatchData.itemsNum /= 2;
-    }
-
-    dispatchData.gws[0] = dispatchData.lws[0];
-    dispatchData.leftovers = dispatchData.dataSetSize % dispatchData.lws[0];
+        // start with 1 thread per data set
+        dispatchData.gws[0] = 1;
+        dispatchData.gws[1] = dispatchData.dataSetsCount;
+        dispatchData.gws[2] = 1;
+        dispatchData.itemsNum = dispatchData.dataSetSize;
+
+        dispatchData.lws[0] = 1;
+        dispatchData.lws[1] = 1;
+        dispatchData.lws[2] = 1;
+        // Compute maximum possible LWS that does not exceed device capabilities and optimizes number of global memory
+        // reads.
+        while ((dispatchData.itemsNum > 32 || dispatchData.lws[0] < dispatchData.itemsNum) && (2 * dispatchData.lws[0] <= max_lws)) {
+            dispatchData.lws[0] *= 2;
+            dispatchData.itemsNum /= 2;
+        }
 
+        dispatchData.gws[0] = dispatchData.lws[0];
+        dispatchData.leftovers = dispatchData.dataSetSize % dispatchData.lws[0];
+    }
     return dispatchData;
 }
 
 JitConstants MVNKernelBfyxOpt::GetJitConstants(const mvn_params& params, MVNKernelBase::DispatchData dispatchData) const {
     auto jit = MVNKernelBase::GetJitConstants(params, dispatchData);
 
-    jit.AddConstants({
-        MakeJitConstant("ITEMS_NUM", dispatchData.itemsNum),
-        MakeJitConstant("LWS", dispatchData.lws[0]),
-        MakeJitConstant("GWS", dispatchData.gws[0]),
-        MakeJitConstant("DATA_SETS_COUNT", dispatchData.dataSetsCount),
-        MakeJitConstant("DATA_SET_SIZE", dispatchData.dataSetSize),
-        MakeJitConstant("LEFTOVERS", dispatchData.leftovers),
-    });
+    if (params.has_dynamic_tensors()) {
+        const auto& input = params.inputs[0];
+        auto x = toCodeString(input.X(), 5);
+        auto y = toCodeString(input.Y(), 4);
+        auto z = toCodeString(input.Z(), 3);
+        auto w = toCodeString(input.W(), 2);
+        auto f = toCodeString(input.Feature(), 1);
+        auto b = toCodeString(input.Batch(), 0);
+        auto multiply = [](std::vector<std::string> dims) -> std::string {
```

Comment:
please create common helper, as this function is being copy-pasted too often (this will be 4th copy).

Reply:
Added common helper function to `jitter` and removed existing redundant code